### ISSUE TITLE
    from mpi4py import MPI          ImportError: DLL load failed: 找不到指定的模块。

### ISSUE THREAD

**Issue by TreeB-450** [agi-brain/xuance#3]
i completely followed the installation, Step1-3,  then pip install xuanpolicy[torch], but when i import xuanpolicy,  it came to "from mpi4py import MPI " and error "ImportError: DLL load failed: 找不到指定的模块。",  What's going on ?

**Comment by wenzhangliu** [comment]
Hi, can you tell me which operating system did you use to install xuanpolicy？

**Comment by TreeB-450** [comment]
> Hi, can you tell me which operating system did you use to install xuanpolicy？

win10

**Comment by TreeB-450** [comment]
> You can try to install an MPI program from [here](https://www.microsoft.com/en-us/download/details.aspx?id=57467). Then, reinstall mpi4py： `pip uninstall mpi4py` `pip install mpi4py`
no, it didn't work.  a new error appear "OSError: [WinError 126] 找不到指定的模块。"

**Comment by wenzhangliu** [comment]
This [link](https://blog.51cto.com/u_14824191/3255805) shows a similar issue to yours, hope it will be useful.

**Comment by blog666** [comment]
![image](https://github.com/agi-brain/xuanpolicy/assets/41368846/afb2b2e1-0064-4ec6-904d-768874a53b57)
按照上述博客方法复制magent.dll到指定文件夹后，还会出现这个错误

**Comment by suishenxiafan** [comment]
Because the software MPI is missing in your PC. For windows system, you can download [msmpisetup.exe] from link https://www.microsoft.com/en-us/download/details.aspx?id=57467 and install it. Then it works.
